### ISSUE TITLE
Handle empty matter

### ISSUE THREAD

**Issue by evilebottnawi** [jonschlinkert/gray-matter#65]
Input:
```
---
---

body{
    border: 1px solid red;
}

```

Maybe return `file.empty` option.
Based on https://github.com/prettier/prettier/issues/4162#issuecomment-373735198

**Comment by thomaslaunch** [comment]
Specifically we would need the front matter to be empty to tell Jekyll that it should process the file. If it is stripped away by formatting then Jekyll copies the file verbatim.

**Comment by jonschlinkert** [comment]
Why not just re-add empty front matter blocks when no data is returned?

**Comment by evilebottnawi** [comment]
@jonschlinkert we have a lot of files, some contain matter blocks some not, we can't add matter blocks always

**Comment by jonschlinkert** [comment]
> we can't add matter blocks always

We could possibly add an option for this, but the implementor (prettier) would still need to allow users to pass options to gray-matter. 

Fwiw, we can't make this default behavior since it's not desirable behavior for any other implementation I've seen besides this use case. 

Here is an example of how I personally would handle it (in prettier I guess, if they decide this should be handled there. if they don't we can discuss other options):

```js
const matter = require('gray-matter');
const file = matter(`
---
---
This is content`);

if (file.matter === '') {
  file.content = file.stringify();
}
console.log(file);
```

**Comment by jonschlinkert** [comment]
actually, we might need to do a little more than what I did in my example since users often add comments to front-matter. Instead of `if (file.matter === ''`) we might need to check length of `file.data` keys or something.

**Comment by evilebottnawi** [comment]
@jonschlinkert we don't add new options, all should work with zero configuration,when we parse files without matter we always have `empty` data, can we just add `property` (example `empty`) when we have empty matter?

**Comment by thomaslaunch** [comment]
A work around is to add nonsense data. But it is nonsense data which is less optimal than leaving the empty blocks.

**Comment by jonschlinkert** [comment]
> can we just add property (example empty) when we have empty matter?

that's a great idea, I'll push something up!

**Comment by jonschlinkert** [comment]
Okay, after playing around with this. I propose we add an `.isEmpty` property on the result. This way you can use the original string if the front-matter is empty. We can also add an `.empty` property, as suggested, but it seems that it would be redundant. 

thoughts?

**Comment by doowb** [comment]
I think the `.isEmpty` property is a good idea. I don't think `.empty` will be necessary.

**Comment by evilebottnawi** [comment]
@jonschlinkert `.isEmpty` is good solution. Can you fix it?

**Comment by jonschlinkert** [comment]
> .isEmpty is good solution.

Great, I'll push it up.

**Comment by evilebottnawi** [comment]
@jonschlinkert friendly ping :+1:

**Comment by jonschlinkert** [comment]
Thanks for the reminder, I'll work on this today!

**Comment by jonschlinkert** [comment]
Published to npm and git tagged as 4.0. `file.isEmpty` is not a boolean. Let me know if you have questions or problems. 

Thanks for the issue
